.github/workflows/codecov.yml

```diff
@@ -0,0 +1,38 @@
+# Based on https://github.com/codecov/codecov-action?tab=readme-ov-file#example-workflowyml-with-codecov-action
+name: Codecov
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+jobs:
+  run:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+    env:
+      OS: ${{ matrix.os }}
+      PYTHON: '3.10'
+    steps:
+    - uses: actions/checkout@main
+    - name: Setup Python
+      uses: actions/setup-python@main
+      with:
+        python-version: '3.10'
```

Comment:
I am fine with having codecov separate (even if that means the unit test run twice, once in this workflow and once in unit tests workflow), but we need to sync python version between them (will start thread on slack)